src/scene_battle_rpg2k3.cpp

```diff
@@ -342,9 +342,11 @@ void Scene_Battle_Rpg2k3::ResetAllBattlerZ() {
 	}
 
 	for (auto* actor: Main_Data::game_party->GetActors()) {
-		auto* sprite = actor->GetBattleSprite();
+		auto* sprite = actor->GetActorBattleSprite();
 		if (sprite) {
 			sprite->ResetZ();
+			sprite->DetectStateChange();
+			sprite->RefreshZ();
```

Comment:
can you put the code of ``RefreshZ`` in ``ResetZ`` or is there a reason why this doesnt work?

Reply:
Unfortunately the code cannot be put into `ResetZ`, because `ResetZ` is in `sprite_battler` while the code has to be run inside of `sprite_actor` because of the `animation` variable.

Reply:
@fmatthew5876 can ResetZ be made virtual or do you have any performance worries.

Reply:
Fixed.

Reply:
`ResetZ()` is called by the constructors of `Sprite_Actor()` and `Sprite_Enemy()`. This means that if someone even subclassed them and tried to override it, the child class function would not get called as the child class hasn't been constructed yet.

If we're going to do this, lets make `ResetZ()` `final` in both of these classes to prevent overriding it.

Reply:
Added a commit now which disallows overriding the `ResetZ()` function in `sprite_actor` and `sprite_enemy`.

Reply:
My bad forgot to check for this annoying limitation :/